packages/um/package.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# Copyright 2024 ACCESS-NRI
+# Based on https://github.com/nci/spack-repo/blob/main/packages/um/package.py
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import configparser
+from spack.package import *
+
+class Um(Package):
+    """
+    UM is a numerical weather prediction and climate modelling software package.
+    """
+
+    homepage = "https://code.metoffice.gov.uk/trac/um"
+    svn = "file:///g/data/ki32/mosrs/um/main/trunk"
+
+    # See 'fcm kp fcm:um.xm' for release versions
+    version("13.0", revision=111272, preferred=True)
+    version("13.1", revision=114076)
+    version("13.2", revision=116723)
+    version("13.3", revision=118802)
+    version("13.4", revision=120750)
+    version("13.5", revision=123226)
+    version("13.6", revision=124981)
+
+    maintainers("penguian")
+
+    variant("optim", default="safe", description="Optimization level",
+        values=("debug", "high", "rigorous", "safe"), multi=False)
+    variant("platform", default="nci-x86-ifort", description="Site platform",
+        values=("nci-x86-ifort", "vm-x86-gnu"), multi=False)
+
+
+    depends_on("fcm", type="build")
+    depends_on("gcom", type=("build", "link"))
+    depends_on("openmpi@4.0.2:", type=("build", "link", "run"))
+    depends_on("eccodes +fortran +netcdf", type=("build", "link", "run"))
+    depends_on("netcdf-fortran@4.5.2", type=("build", "link", "run"))
+
+    phases = ["build", "install"]
+
+    # The dependency name and the ld_flags from
+    # the FCM config for each library configured via FCM.
+    _lib_cfg = {
+        "DR_HOOK": {
+            "dep_name": "drhook",
+            "fcm_ld_flags": "-ldrhook"},
+        "eccodes": {
+            "dep_name": "eccodes",
+            "fcm_ld_flags": "-leccodes_f90 -leccodes"},
+        "netcdf": {
+            "dep_name": "netcdf-fortran",
+            "fcm_ld_flags": "-lnetcdff -lnetcdf"}}
+
+
+    def _get_linker_args(self, spec, fcm_libname):
+        """
+        The reason for the explicit -rpath is:
+        https://github.com/ACCESS-NRI/spack_packages/issues/14#issuecomment-1653651447
+        """
+        dep_name = self._lib_cfg[fcm_libname]["dep_name"]
+        ld_flags = spec[dep_name].libs.ld_flags
+        fcm_ld_flags = self._lib_cfg[fcm_libname]["fcm_ld_flags"]
+        rpaths = ["-Wl,-rpath=" + d for d in spec[dep_name].libs.directories]
+        return " ".join([ld_flags, fcm_ld_flags] + rpaths)
```

Comment:
Check if there's a syntax error.

Reply:
There is no syntax error. All check have passed: `ld_flags` and `fcm_ld_flags` are both strings, while `rpaths` is a list of strings.